### ISSUE TITLE
emit TaskEvent for task stops blocked by `shutdown_delay`

### ISSUE THREAD

**Issue by tgross** [hashicorp/nomad#14170]
When a task shuts down with a `shutdown_delay` set, there's no Task Event that indicates we're waiting for the task to shut down. It can be useful for debugging what would otherwise look like a "stuck" task to have this in the output of `nomad alloc status` (and the corresponding pages on the UI).

We've already got a debug-level log entry at the right spot in [`task_runner.go#L931`](https://github.com/hashicorp/nomad/blob/v1.3.3/client/allocrunner/taskrunner/task_runner.go#L931), so it would be a matter of creating a new Task Event and calling `UpdateState` for it (similar to [`task_runner.go#L1143-L1145`](https://github.com/hashicorp/nomad/blob/v1.3.3/client/allocrunner/taskrunner/task_runner.go#L1143-L1145)).

**Comment by DarthBenro008** [comment]
Hi @tgross , I'd be happy to send a PR for the same, thanks for the detailed description! :D

**Comment by cesarvspr** [comment]
@DarthBenro008 I would be happy to assist if needed, please share your PR with us

**Comment by DarthBenro008** [comment]
@cesarvspr I have raised a PR!
